lib/netdev-rte-offloads.c

```
@@ -1072,8 +1075,7 @@ netdev_rte_offloads_flow_put(struct netdev *netdev, struct match *match,
 
     if (ufid_hw_offload) {
         VLOG_DBG("got modification and destroying previous rte_flow");
-        ufid_hw_offload_remove(ufid, &rte_port->ufid_to_rte);
-        ret = netdev_rte_port_ufid_hw_offload_free(ufid_hw_offload);
+        ret = netdev_offloads_flow_del(ufid);
```

Function netdev_offloads_flow_del() always return positive errors. Therefore the test 
``
if (ret < 0)
`` is non effective. More than that - the function netdev_offloads_flow_del() ignores any errors from netdev_rte_port_ufid_hw_offload_free(). It must be fixed to consider those errors.

---

right. though netdev_rte_port_ufid_hw_offload_free returns always 0...
it should be done in another commit though, maybe in other placed in the code as well.
i'd suggest to comply with the standard and return negative value as an error